### PR TITLE
Refactor unit conversion

### PR DESCRIPTION
**Pull request by javerbukh**
This PR has the UI and basic functionality for unit conversion, but I still have questions as to using Signals for data conversion and where the data will be stored.

### CODE REVIEW

File: specviz/widgets/plotting.py [hunk 1]
```diff
@@ -503,3 +534,270 @@ def _on_remove_linear_region(self):
...
+        self._data_unit = data_unit
+        self._spectral_axis_unit = spectral_axis_unit
+
+        self._plot_item.setLabel('bottom', units=self.spectral_axis_unit)
+        self._plot_item.setLabel('left', units=self.data_unit)
+
+        self.autoRange()
+
+
+class UnitChangeDialog(QDialog):
```
Comment (nmearl) [inline specviz/widgets/plotting.py]: The `UnitChangeDialog` should be defined in its own file, probably in the `specviz/widgets` directory.

File: specviz/widgets/workspace.py [hunk 2]
```diff
@@ -132,6 +132,17 @@ def _on_toggle_visibility(self, state):
 
         self.proxy_model.dataChanged.emit(idx, idx)
 
+    def _on_changed_color(self, color):
```
Comment (nmearl) [inline specviz/widgets/workspace.py]: What is this method used for? There already is the method in `plotting.py` that handles changing the line color.

**Comment by nmearl** [comment]
Could you remove the changes to the `astropy_helpers` -- those are handled separately.

**Comment by javerbukh** [comment]
@nmearl I'm not sure how to remove the changes to astropy_helpers. I did not make any knowing changes to it. I've made the other changes now and will push a new commit.

**Comment by javerbukh** [comment]
@nmearl This PR is ready for review. Unit conversion works for spectral axis units, but when it comes to data units, it's hard to find something that Jansky's will convert to. That being said, everything is functionally and visually there.

**Comment by javerbukh** [comment]
@nmearl Alright, I ran that command on workspace.py and astropy_helpers and removed the .pytest_cache/. Let me know how it looks.

**Pull request merged**